blazor/datagrid/column-menu.md

```diff
@@ -70,6 +70,87 @@ The default menu items are displayed in the following table,
+            <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Center" Width="120"></GridColumn>
+        </GridColumns>
+    </SfGrid>
+}
+else if (OrderIDFreezeDirection == FreezeDirection.Right)
+{
+    <SfGrid @ref="Grid" DataSource="@Orders" Height="315" ColumnMenuItems="@(new List<ColumnMenuItemModel>() { new ColumnMenuItemModel { Text = "Change Freeze Direction", Id = "OrderID" } })" ShowColumnMenu="true" AllowPaging="true">
+        <GridEvents ColumnMenuItemClicked="ColumnMenuItemClickedHandler" TValue="Order"></GridEvents>
+        <GridColumns>
+            <GridColumn Field=@nameof(Order.OrderID) Freeze="@OrderIDFreezeDirection" IsFrozen="true" HeaderText="Order ID" TextAlign="TextAlign.Center" Width="120"></GridColumn>
+            <GridColumn Field=@nameof(Order.CustomerID) HeaderText="Customer Name" Width="150" TextAlign="TextAlign.Center" ShowColumnMenu="false"></GridColumn>
+            <GridColumn Field=@nameof(Order.OrderDate) HeaderText="Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Center" Width="130"></GridColumn>
+            <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Center" Width="120"></GridColumn>
+        </GridColumns>
+    </SfGrid>
+}
+
+@code{
+    public string[] MenuItems = new string[] { "Group", "Ungroup", "ColumnChooser", "Filter" };
+    SfGrid<Order> Grid { get; set; }
+    public List<Order> Orders { get; set; }
+    public FreezeDirection OrderIDFreezeDirection { get; set; } = FreezeDirection.Left;
+
+    public void ColumnMenuItemClickedHandler(ColumnMenuClickEventArgs args)
+    {
+        if (args.Item.Id == "OrderID")
+        {
+            if (OrderIDFreezeDirection == FreezeDirection.Left)
+            {
+                OrderIDFreezeDirection = FreezeDirection.Right;
+            }
+            else
+            {
+                OrderIDFreezeDirection = FreezeDirection.Left;
+            }
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        Orders = Enumerable.Range(1, 75).Select(x => new Order()
+        {
+            OrderID = 1000 + x,
+            CustomerID = (new string[] { "ALFKI", "ANANTR", "ANTON", "BLONP", "BOLID" })[new Random().Next(5)],
+            Freight = 2.1 * x,
+            OrderDate = DateTime.Now.AddDays(-x),
+        }).ToList();
+    }
+
+    public class Order
+    {
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public DateTime? OrderDate { get; set; }
+        public double? Freight { get; set; }
+    }
+}
+
+```
+
+![Customize Column Menu in Blazor DataGrid](images/blazor-datagrid-custom-column-menu.gif)
+
+> [View Sample in GitHub.](https://github.com/SyncfusionExamples/blazor-datagrid-customize-column-menu)
```

Comment:
and the gif image is not proper. It shows like misalignment in header and content

Reply:
new gif image added